File: 29-divide-two-integers/divide-two-integers.py
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        if (dividend == -2147483648 and divisor == -1) or (dividend == -2147483648 and divisor == -1):
            return 2147483647
        if (dividend > 0 and divisor > 0) or (dividend < 0 and divisor < 0):
            return math.floor(dividend / divisor)
        if (dividend > 0 and divisor < 0) or (dividend < 0 and divisor > 0):
            return math.ceil(dividend / divisor)
        if dividend == divisor == -1:
            return 1
        if dividend == 0:
            return 0
        
        
        

File: 29-divide-two-integers/divide-two-integers.java
class Solution {
    public int divide(int dividend, int divisor) {
        if ((dividend == Integer.MIN_VALUE && divisor == -1) || 
            (dividend == Integer.MIN_VALUE && divisor == -1)) {
            return Integer.MAX_VALUE;
        }

        if ((dividend > 0 && divisor > 0) || (dividend < 0 && divisor < 0)) {
            return (int) Math.floor((double) dividend / divisor);
        }
        
        if ((dividend > 0 && divisor < 0) || (dividend < 0 && divisor > 0)) {
            return (int) Math.ceil((double) dividend / divisor);
        }

        return 0;
    }
}
